09_class_bank_card.py

```diff
@@ -0,0 +1,178 @@
+        self.__pin = num
+
+    @property
+    def balance(self):
+        """Getter for credit card balance."""
+        return format(self.__balance, '.2f')
+
+    def check_pin_code(self):
+        """Check if pin code is correct."""
+        current_pin = int(input('Please enter your PIN: \n'))
+        if current_pin == self.__pin:
+            return True
+        else:
+            return False
+
+    def check_cvv_code(self):
+        """Check if cvv code is correct."""
+        current_cvv = int(input('Please enter your cvv code: \n'))
+        if current_cvv == self.__cvv:
+            return True
+        else:
+            return False
+
+    def check_card_number(self):
+        """Check if card number is correct."""
+        current_card_num = int(input('Please enter your card number: \n'))
+        if current_card_num == self.__card_num:
+            return True
+        else:
+            return False
+
+    def validate_expired_card(self, end_date):
+        """Check the card expiration date."""
+        if self.validate_date(end_date):
+            return True
+
+    def __income(self, new_sum, usd=False):
+        """Balance increase.
+
+        params new_sum: An integer representing the amount of money.
+        params usd: A bool representing if sum in usd.
+        """
+        operation = self._uah_to_usd(new_sum) if usd else new_sum
+        self.__balance += operation
+        return self.__balance
+
+    def __expenses(self, new_sum, usd=False):
+        """Decrease in balance.
+
+        params new_sum: An integer representing the amount of money.
+        params usd: A bool representing if sum in usd.
+        """
+        operation = self._uah_to_usd(new_sum) if usd else new_sum
+        self.__balance -= operation
+        return self.__balance
+
+    def check_balance(self):
+        """Check balance in credit card."""
+        if self.check_pin_code():
+            return (f'Hello {self.__name} {self.__surname}.'
+                    f'Your balance is {self.balance} UAH')
+        else:
+            return "PIN isn't correct"
+
+    def put_money_to_card(self, usd=False):
+        """Top up credit card balance.
+
+        params usd: A bool representing if sum in usd.
+        """
+        if self.check_card_number():
+            cur_sum = int(input('Enter amount: \n'))
+            self.__income(cur_sum, usd=True) if usd else self.__income(cur_sum)
+            return 'Operation was successfully completed'
+        else:
+            return "Card number isn't correct"
+
+    def buy_new_item(self, num, usd=False):
+        """Reducing the balance on the card after purchasing an item.
+
+        params num: An integer representing the amount of money.
+        params usd: A bool representing if sum in usd.
+        """
+        if self.check_cvv_code():
+            if self.validate_expired_card(self.__end_date):
+                self.__expenses(num, usd=True) if usd else self.__expenses(num)
```

Comment:
We have ternary if. 
And if usd = True in args of method buy_new_item
Then we run self.__expenses with usd=True
But if usd = False, we run self.__expenses without usd=True, because in args of method buy_new_item default value usd is False